src/gui/src/layoutViewer.cpp

```diff
@@ -2485,8 +2504,24 @@ void LayoutViewer::updateBlockPainting(const QRect& area)
   // paint layout
   drawBlock(&block_painter, dbu_bounds, 0);
 
+  if (showFps_()) {
+    now = std::chrono::system_clock::now();
+    auto frame_time_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
+                            now - last_paint_time_)
+                            .count();
+    QFont fps_font;
+    fps_font.setPixelSize(25);
+    fps_font.setBold(true);
+    block_painter.resetTransform();
+    block_painter.setFont(fps_font);
+    block_painter.setPen(Qt::yellow);
+    block_painter.translate(10, 40);
+    block_painter.drawText(0, 0,
+                          "fps: " + QString::number(1000.0 / frame_time_ms));
+  }
```

Comment:
FPS is not really what you are measuring, since we do not force repaints on every possible attempt. I like the idea of having this available, but might be good to instrument the code with a "debug" mode instead that can provide an overlay with the paint times for major blocks of code (pins, instances, layers, etc.) so we can better isolate what improvements can be made. I would just use the paint time instead of FPS.